### PR TITLE
Add docs for hash join v2

### PR DESCRIPTION
**Pull request by windtalker**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -3248,6 +3248,21 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+> The feature controlled by this variable is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `legacy`
+- Possible values: `legacy`, `optimized`
+- This variable is used to control whether use optimized hash join in TiDB. The value is `legacy` by default. If it is set to `optimized`, TiDB will use optimized hash join to execute hash join, which is expected to have better performance.
+- Currently optimized hash join only support inner join and outer join, so for other joins, even if `tidb_hash_join_version` is set to `optimized`, it will still use the legacy hash join.
```
Comment (qiancai) [inline system-variables.md:3264]:
```suggestion
- This variable controls whether TiDB uses an optimized method for executing the hash join operator. The default value is `legacy`, which means the optimized method is not used. When this variable is set to `optimized`, TiDB uses the optimized method for executing the hash join operator to improve hash join performance.
- Currently, only inner join and outer join operations support the optimized method for hash join. For other types of join operations, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the legacy method when executing the Hash Join operator.
```

Comment (windtalker) [reply system-variables.md:3264]: lgtm

File: system-variables.md [hunk 2]
```diff
@@ -3248,6 +3248,26 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `legacy`
+- Possible values: `legacy`, `optimized`
+- This variable is used to control whether use optimized hash join in TiDB. The value is `legacy` by default. If it is set to `optimized`, TiDB will use optimized hash join to execute hash join, which is expected to have better performance.
+
+> **Note:**
+>
+> - Currently optimized hash join only support inner join and outer join, so for other joins, even if `tidb_hash_join_version` is set to `optimized`, it will still use the legacy hash join.
+> - Currently optimized hash join does not support spill to disk when memory is out of quota.
```
Comment (qiancai) [inline system-variables.md:3268]:
```suggestion
- This variable is used to control whether TiDB uses an optimized version of hash join. The value is `legacy` by default, which means the optimized version is not used. If it is set to `optimized`, TiDB uses the optimized version to execute hash join for better performance.

> **Note:**
>
> - Currently, the optimized hash join only supports inner join and outer join, so for other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the legacy hash join.
> - Currently, the optimized hash join does not support spill to disk when the memory is out of quota.
```

Comment (windtalker) [reply system-variables.md:3268]: lgtm

File: system-variables.md [hunk 3]
```diff
@@ -3248,6 +3248,26 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+
+
```
Comment (Oreoxmt) [inline system-variables.md:3270]:
```suggestion
```

File: system-variables.md [hunk 4]
```diff
@@ -3248,6 +3248,26 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `legacy`
+- Possible values: `legacy`, `optimized`
+- This variable is used to control whether TiDB uses an optimized version of hash join. The value is `legacy` by default, which means the optimized version is not used. If it is set to `optimized`, TiDB uses the optimized version to execute hash join for better performance.
+
+> **Note:**
+>
+> - Currently, the optimized hash join only supports inner join and outer join, so for other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the legacy hash join.
+> - Currently, the optimized hash join does not support spill to disk when the memory is out of quota.
```
Comment (Oreoxmt) [inline system-variables.md:3268]:
```suggestion
> - Currently, the optimized hash join does not support spilling to disk when the memory is out of quota.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**